src/main/java/org/simulator/math/QualityMeasure.java

```diff
@@ -160,14 +161,21 @@ public double distance(MultiTable x, MultiTable expected) {
 
     // Get results by calling to newly added getAbsDistances method
     ArrayList<Double> pp = getMaxAbsDistances(left, right);
-    System.out.println("MaxAbsDistanceResults: " + meanFunction.computeMean(pp));
+    System.out.println(pp);
+    System.out.println("MaxAbsDistanceResults: " + Collections.max(pp));
 
     ArrayList<Double> distances = new ArrayList<Double>();
     distances.addAll(getColumnDistances(left, right));
     System.out.println("Current Distance Function results: " + meanFunction.computeMean(distances));
     return meanFunction.computeMean(distances);
   }
 
+  /**
+   *
+   * @param x
+   * @param expected
+   * @return maxAbsDistances the list of maximum absolute distances in the blocks
+   */
   public ArrayList<Double> getMaxAbsDistances(MultiTable x, MultiTable expected) {
```

Comment:
Please do not declare unnecessarily specific return types for methods and also declare variables with the least specific type. In this concrete case, please use `List<Double>` to initialize the variable for the `distances` and also return a `List<Double>`. You can still use an `ArrayList<Double>` to initialize the variable, but not as return type. This applies to other positions as well.